### PR TITLE
Test configure_apiserver while addressing issue with authorization_modes

### PR DESCRIPTION
**Pull request by addyess**
## Overview 
Addresses [LP#2071488](https://launchpad.net/bugs/2071488) by not assigning the output of `list.remove` back to the list.  python's `list.remove()` always returns None

## Details
* ensures that this method that is changed is covered by SOME unit tests
* makes the code more testable
* fixes the issue by not assigning None to the list i meant to remove an item from

### CODE REVIEW

File: charms/kubernetes_snaps.py [hunk 1]
```diff
@@ -135,7 +135,7 @@ def configure_apiserver(
                 "Authorization mode includes 'Webhook' but no webhook config file is present."
                 "'Webhook' must be removed from authorization mode."
             )
-            authorization_modes = authorization_modes.remove("Webhook")
```
Comment (addyess) [inline charms/kubernetes_snaps.py:138]: This is the original thrust of the bug.  Python's list.remove returns None.  causing the following stack track:

```
  File "/var/lib/juju/agents/unit-kubernetes-control-plane-0/charm/./src/charm.py", line 541, in reconcile
    self.configure_apiserver()
  File "/var/lib/juju/agents/unit-kubernetes-control-plane-0/charm/./src/charm.py", line 141, in configure_apiserver
    kubernetes_snaps.configure_apiserver(
  File "/var/lib/juju/agents/unit-kubernetes-control-plane-0/charm/venv/charms/kubernetes_snaps.py", line 145, in configure_apiserver
    api_opts["authorization-mode"] = ",".join(authorization_modes)
TypeError: can only join an iterable
```

File: charms/kubernetes_snaps.py [hunk 2]
```diff
@@ -163,28 +163,28 @@ def configure_apiserver(
...
-    os.makedirs(audit_root, exist_ok=True)
+    audit_root = Path("/root/cdk/audit")
+    audit_log_path = audit_root / "audit.log"
+    audit_policy_path = audit_root / "audit-policy.yaml"
+    audit_webhook_conf_path = audit_root / "audit-webhook-config.yaml"
+    audit_root.mkdir(exist_ok=True)
 
-    api_opts["audit-log-path"] = audit_log_path
+    api_opts["audit-log-path"] = str(audit_log_path)
     api_opts["audit-log-maxage"] = "30"
```
Comment (addyess) [inline charms/kubernetes_snaps.py:173]: These make the method more testable

File: charms/kubernetes_snaps.py [hunk 3]
```diff
@@ -163,28 +163,28 @@ def configure_apiserver(
...
     else:
         remove_if_exists(audit_policy_path)
 
     if audit_webhook_conf:
-        with open(audit_webhook_conf_path, "w") as f:
+        with audit_webhook_conf_path.open("w") as f:
             f.write(audit_webhook_conf)
-        api_opts["audit-webhook-config-file"] = audit_webhook_conf_path
+        api_opts["audit-webhook-config-file"] = str(audit_webhook_conf_path)
     else:
```
Comment (addyess) [inline charms/kubernetes_snaps.py:188]: again -- more test-ability -- only need to mock out `Path`

File: tests/unit/test_kubernetes_snaps.py [hunk 4]
```diff
@@ -138,3 +137,75 @@ def test_configure_kubelet(
     if external_cloud.has_xcp:
         expected_args["cloud-provider"] = "external"
     assert expected_args == args
+
+
+@mock.patch("charms.kubernetes_snaps.configure_kubernetes_service")
+@mock.patch("charms.kubernetes_snaps.Path")
+def test_configure_apiserver(mock_path, configure_kubernetes_service, external_cloud):
```
Comment (eaudetcobello) [inline tests/unit/test_kubernetes_snaps.py:144]: Nice! Useful test.

**Review by eaudetcobello — APPROVED** [review]

**Pull request merged**